src/sync/app.cpp

```diff
@@ -734,6 +730,112 @@ void App::link_user(std::shared_ptr<SyncUser> user,
     
     App::log_in_with_credentials(credentials, user, completion_block);
 }
+    
+    void App::refresh_custom_data(std::function<void(Optional<AppError>)> completion_block)
+    {
+        refresh_access_token(completion_block);
+    }
+    
+    void App::do_authenticated_request(Request request,
+                                       std::function<void (Response)> completion_block) const
+    {
+        auto handler = [completion_block, request, this](const Response& response) {
+            if (auto error = check_for_errors(response)) {
+                App::handle_auth_failure(error.value(), response, request, completion_block);
+            } else {
+                completion_block(response);
+            }
+        };
+
+        request.headers = get_request_headers(current_user(),
```

Comment:
It looks a bit dangerous to use the current user here. I would much prefer it if it was parsed in as a parameter.

Generally, I think there is a very high chance we introduce subtle race conditions by using `current_user()` inside functions, especially when it involves callbacks to bindings which might delegate work to other threads.

IMO, we should parse in the user as a parameter where it is needed. It will not only make it easier to reason about but will also enable more flexible API's in the SDK's.

Reply:
Agreed, added the ability to now pass user as a param